src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java

```diff
@@ -232,6 +233,39 @@ public Response addDataverse(@Context ContainerRequestContext crc, String body,
 
         }
     }
+    
+    @POST
+    @AuthRequired
+    @Path("{identifier}/validateDatasetJson")
+    @Consumes("application/json")
+    public Response validateDatasetJson(@Context ContainerRequestContext crc, String body, @PathParam("identifier") String idtf) {
+        User u = getRequestUser(crc);
+        try {
+            String validationMessage = execCommand(new ValidateDatasetJsonCommand(createDataverseRequest(u), findDataverseOrDie(idtf), body));
+            return ok(validationMessage);
+        } catch (WrappedResponse ex) {
+            Logger.getLogger(Dataverses.class.getName()).log(Level.SEVERE, null, ex);
+            return ex.getResponse();
+        }
+    }
+    
+    @GET
+    @AuthRequired
+    @Path("{identifier}/datasetSchema")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getDatasetSchema(@Context ContainerRequestContext crc, @PathParam("identifier") String idtf) {
+        User u = getRequestUser(crc);
+
+        try {
+            String datasetSchema = execCommand(new GetDatasetSchemaCommand(createDataverseRequest(u), findDataverseOrDie(idtf)));
+            return ok(datasetSchema);
```

Comment:
@sekmiller and were talking abouthow we probably want to return just the JSON instead of escaped JSON in our normal "ok... data" data structure.

Jim did this recently in this commit:  9953 - don't wrap linkset in a data element 3a4d8f98053ff726c617a45c5ad15d2f3059c138

Reply:
@sekmiller and I decided to return just the JSON Schema. See 2d3f7ab. This way instead of the \n characters and data...message, you get just what you want:

```
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "$defs": {
        "field": {
            "type": "object",
            "required": [
                "typeClass",
...
```